flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/FileSourceSplit.java

```diff
@@ -91,9 +98,17 @@
      * @param filePath The path to the file.
      * @param offset The start (inclusive) of the split's rage in the file.
      * @param length The number of bytes in the split (starting from the offset)
+     * @param modificationTime The modification time of the file
+     * @param fileSize The size of the full file
      */
-    public FileSourceSplit(String id, Path filePath, long offset, long length) {
-        this(id, filePath, offset, length, NO_HOSTS);
+    public FileSourceSplit(
```

Comment:
Strictly speaking, we cannot easily do this change because the class is annotated with `@PublicEvolving` and the change might break existing setups.

We definitely have to keep the old constructors in case someone is using them.